lib/deploy_pin/collector.rb

```diff
@@ -56,5 +40,27 @@ def exacutable
         task if _tasks[0..index].none? { |_task| task.eql?(_task) }
       end.compact
     end
+
+    def tasks_count
+      tasks.count
+    end
+
+    private
+
+      def files
+        Dir["#{DeployPin.tasks_path}/*.rb"]
+      end
+
+      def tasks
+        files.map do |file|
+          task = DeployPin::Task.new(file)
+          task.parse_file
+
+          # if task is in specified group and it is not done or if run by uuid
+          next unless (identifiers.include?(task.group) && !task.done?) || identifiers.include?(task.uuid)
```

Comment:
If you want to be able rerun task which is already done, it is better to control that by exclamation mark at the end of uuid parameter. To be confident that a task is executed one more time safely. Also, it would be great to clean up duplicated record in the db